### ISSUE TITLE
Interaction.defer is broken after #2579

### ISSUE THREAD

**Issue by tibue99** [Pycord-Development/pycord#2682]
### Summary

#2579 breaks `interaction.defer()`

### Reproduction Steps

Run the code below or any other code that uses `interaction.defer()` 

### Minimal Reproducible Code

```python
import discord

bot = discord.Bot()


@bot.slash_command()
async def hey(ctx):
    await ctx.defer()
    await ctx.respond("Hey!")

bot.run("")
```

### Expected Results

No error

### Actual Results
```
Ignoring exception in command hey:
Traceback (most recent call last):
  File "C:\Users\Timo\Desktop\pythonProject\.venv\Lib\site-packages\discord\commands\core.py", line 138, in wrapped
    ret = await coro(arg)
          ^^^^^^^^^^^^^^^
  File "C:\Users\Timo\Desktop\pythonProject\.venv\Lib\site-packages\discord\commands\core.py", line 1082, in _invoke
    await self.callback(ctx, **kwargs)
  File "C:\Users\Timo\Desktop\pythonProject\main.py", line 8, in hey
    await ctx.defer()
  File "C:\Users\Timo\Desktop\pythonProject\.venv\Lib\site-packages\discord\interactions.py", line 826, in defer
    adapter.create_interaction_response(
  File "C:\Users\Timo\Desktop\pythonProject\.venv\Lib\site-packages\discord\webhook\async_.py", line 530, in create_interaction_response
    payload["data"]["attachments"] = attachments
    ~~~~~~~^^^^^^^^
KeyError: 'data'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Timo\Desktop\pythonProject\.venv\Lib\site-packages\discord\bot.py", line 1149, in invoke_application_command
    await ctx.command.invoke(ctx)
  File "C:\Users\Timo\Desktop\pythonProject\.venv\Lib\site-packages\discord\commands\core.py", line 435, in invoke
    await injected(ctx)
  File "C:\Users\Timo\Desktop\pythonProject\.venv\Lib\site-packages\discord\commands\core.py", line 146, in wrapped
    raise ApplicationCommandInvokeError(exc) from exc
discord.errors.ApplicationCommandInvokeError: Application Command raised an exception: KeyError: 'data'
```

### Intents

-

### System Information

- Python v3.12.1-final
- py-cord v2.6.1-final
- aiohttp v3.11.11
- system info: Windows 11 10.0.26100

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

**Comment by Paillat-dev** [comment]
I wonder why using pydantic would be a good idea... Anyhow. I'll see if I can pr a fix unless @Icebluewolf wants to do it since its his pr

**Comment by Icebluewolf** [comment]
It's actually #2679 that breaks it. Will fix soon.

**Comment by Lulalaby** [comment]
should i revert for now?

**Comment by Paillat-dev** [comment]
#2679 is not merged, how would it cause an issue on master ?

**Comment by Icebluewolf** [comment]
I "fixed" #2679's issue while writing #2579 and the "fix" never got removed from #2579 when I split it into 2 PRs.

**Comment by Dorukyum** [comment]
Can reproduce.

**Comment by Dorukyum** [comment]
Fixed in https://github.com/Pycord-Development/pycord/commit/1ca07cb133264ee81db1051cf171ce6f528570eb.
